kv/buffer_store.go

```diff
@@ -66,6 +66,19 @@ func (s *BufferStore) Seek(k Key) (Iterator, error) {
 	return newUnionIter(bufferIt, retrieverIt), nil
 }
 
+// SeekReverse implements the Retriever interface.
+func (s *BufferStore) SeekReverse(k Key) (Iterator, error) {
+	bufferIt, err := s.MemBuffer.SeekReverse(k)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	retrieverIt, err := s.r.SeekReverse(k)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return newUnionIter(bufferIt, retrieverIt), nil
```

Comment:
I think we should update unionIter too. The merge algorithm depends on key comparison.